fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBSimpleQueryGraphTest.java

```diff
@@ -188,5 +191,78 @@ public void testSimplePlanGraph() throws Exception {
+                    return GroupExpressionRef.of(new LogicalSortExpression(null, false, qun));
+                },
+                Optional.of(ImmutableSet.of("RestaurantRecord")),
+                Optional.empty(),
+                IndexQueryabilityFilter.TRUE,
+                false,
+                ParameterRelationshipGraph.empty()));
+        Assertions.assertEquals("Cascades planner could not plan query", exception.getMessage());
+
+        // with index hints, plan a different query
+        final var plan2 = cascadesPlanner.planGraph(
+                () -> {
+                    var qun =
+                            Quantifier.forEach(GroupExpressionRef.of(
+                                    new FullUnorderedScanExpression(ImmutableSet.of("RestaurantRecord",
+                                            "RestaurantReviewer"), ImmutableSet.of("RestaurantRecord$name"))));
+
+                    qun = Quantifier.forEach(GroupExpressionRef.of(
+                            new LogicalTypeFilterExpression(ImmutableSet.of("RestaurantRecord"),
+                                    qun,
+                                    Type.Record.fromDescriptor(TestRecords4Proto.RestaurantRecord.getDescriptor()))));
+
+                    final var graphExpansionBuilder = GraphExpansion.builder();
+
+                    graphExpansionBuilder.addQuantifier(qun);
+                    final var nameValue =
+                            new FieldValue(qun.getFlowedObjectValue(), ImmutableList.of("name"));
+                    final var restNoValue =
+                            new FieldValue(qun.getFlowedObjectValue(), ImmutableList.of("rest_no"));
+
+                    graphExpansionBuilder.addPredicate(new ValuePredicate(restNoValue, new Comparisons.SimpleComparison(Comparisons.Type.GREATER_THAN, 1L)));
+                    graphExpansionBuilder.addResultColumn(Column.of(Type.Record.Field.of(nameValue.getResultType(), Optional.of("nameNew")), nameValue));
+                    graphExpansionBuilder.addResultColumn(Column.of(Type.Record.Field.of(restNoValue.getResultType(), Optional.of("restNoNew")), restNoValue));
+                    qun = Quantifier.forEach(GroupExpressionRef.of(graphExpansionBuilder.build().buildSelect()));
+                    return GroupExpressionRef.of(new LogicalSortExpression(null, false, qun));
+                },
+                Optional.of(ImmutableSet.of("RestaurantRecord")),
+                Optional.empty(),
+                IndexQueryabilityFilter.TRUE,
+                false,
+                ParameterRelationshipGraph.empty());
+
+        Assertions.assertFalse(mapPlan(
+                typeFilterPlan(
+                        scanPlan()
+                                .where(scanComparisons(range("([1],>")))))
+                .where(result(recordConstructorValue(exactly(fieldValue("name"), fieldValue("rest_no"))))).matchesExactly(plan2));
```

Comment:
This tests that a different plan is generated, how to construct the plan that can assertMatchExactly this new plan?